snow/engine/snowman/transitive.go

```diff
@@ -470,39 +470,79 @@ func (t *Transitive) Start(ctx context.Context, startReqID uint32) error {
 		return err
 	}
 
+	preferredID := t.VM.GetInitialPreference()
+	preferredBlock, err := t.VM.GetBlock(ctx, preferredID)
+	if err != nil {
+		return err
+	}
+
+	// latest preference
+	preferredTip := lastAcceptedID
+	if preferredBlock.Height() > lastAccepted.Height() {
+		preferredTip = preferredID
+	}
+
+	preferredChain := make([]snowman.Block, 0)
+	for preferredBlock.Status() == choices.Processing {
+		// During bootstrap our accepted tip can grow, so we should only
+		// re-issue blocks into consensus with a greater height than what
+		// we currently have.
+		if preferredBlock.Height() > lastAccepted.Height() {
+			// Insert the most recently preferred blocks at the head of the
+			// queue, so we can insert them into consensus in-order later.
+			preferredChain = append([]snowman.Block{preferredBlock}, preferredChain...)
+		}
```

Comment:
If we want the for loop below to iterate forward instead of iterating backwards, could we insert in reverse order one by one and then reverse at the end?

Very unlikely this is a needed performance improvement, but I think it is also more idiomatic to do:

```go
preferredChain = append(preferredChain, preferredBlock)
```

Reply:
Just found out `slices.Reverse` exists ^_^